### PR TITLE
Fix null client ID used in exception report and spelling errors, fixes #437

### LINKED ISSUE #437: null Client ID shown in log file if checkoutRun fails to get files for a run
Thank you for working to help us improve PC^2!  
Please fill in the following fields in order to help us understand what issue you are reporting.

**Describe the issue**: 
The log shows:
220527 141857.479|INFO|Thread-158525|checkoutRun|runUnavailableException Client null can not checked out run 1340 (site 1)

"Client null" in incorrect.  The wrong variable is used when throwing the exception:
               throw new RunUnavailableException("Client " + clientId + " can not checked out run " + run.getNumber() + " (site " + run.getSiteNumber() + ")");
It should be "whoChangedRun" and not clientId.

**To Reproduce**: 
Hard to reproduce.  You have to delete the relevant db files for the run you want to test with.  Then, rejudge or view the run.
The exception will occur if the run files can't be found.

**Expected behavior**: 
The correct client ID of the client fetching the run should be displayed, not 'null'.

**Actual behavior**: 
'null' is displayed for the Client ID of the client requesting the run.

**Environment**: please complete the following information:
 All

**Log Info**:  
  220527 141857.478|INFO|Thread-158478|info|sendToClient b4 to JUDGE10 @ site 1 Packet[RUN_NOTAVAILABLE] #22332 SERVER0 @ site 1 -> JUDGE10 @ site 1
220527 141857.479|INFO|Thread-158525|checkoutRun|runUnavailableException Client null can not checked out run 1340 (site 1)
22

**Screenshots**:  i

**Additional context**:
This was noticed in the Server0@site1-1.log file for the ACPC kick-off contest in June 2022.

### PR DESCRIPTION
**Pull request by johnbrvc**
At the point the exception is thrown, the original clientId variable was always null.  It should have been the client who requested the checkout of the run, namely, whoChangedRun.  The same problem existed for checking out Clarifications, and that too has been fixed (in the same module).
In addition, there was a wrong word used in the string text that was thrown:  "can not checked out" changed to "can not check out".

Before submitting your Pull Request, please make sure you have read the [Guidelines for Submitting Pull Requests](https://github.com/pc2ccs/pc2v9/wiki/Guidelines-for-Submitting-Pull-Requests).  Then please provide the following information:

### Description of what the PR does
Corrects incorrect variable usage in a "throw" which would cause "null" to be part of the exception string when the intent was to show the information for the client making the checkout request. 
Corrects the spelling/wrong word usage in the exception strings.

### Issue which the PR fixes
#437 
### Environment in which the PR was developed (OS,IDE, Java version, etc.)
OS              : Windows 11 10.0 (amd64)
Java Version    : 1.8.0_321

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)
This fix can not easily be tested without "breaking" the code to cause the exception to be thrown.  The code was recently fixed to prevent this very exception.  In order to test, in InternalContest.checkoutRun(), change the "if(canBeCheckedOut)" to "if(!canBeCheckedOut)".  This will force the exception.  The log message shown is now:

**220613 133956.215|INFO|Thread-9|checkoutRun|runUnavailableException Client ADMINISTRATOR1 @ site 1 can not check out run 1359 (site 1)**

(Be sure to undo the "breaking" of the code to test this!)

### CODE REVIEW

File: src/edu/csus/ecs/pc2/core/model/InternalContest.java [hunk 1]
```diff
@@ -1493,7 +1493,7 @@ public Run checkoutRun(Run run, ClientId whoChangedRun, boolean reCheckoutRun, b
                 runList.updateRun(newRun);
                 return runList.get(run.getElementId());
             } else {
-                throw new RunUnavailableException("Client " + clientId + " can not checked out run " + run.getNumber() + " (site " + run.getSiteNumber() + ")");
+                throw new RunUnavailableException("Client " + whoChangedRun + " can not check out run " + run.getNumber() + " (site " + run.getSiteNumber() + ")");
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/model/InternalContest.java:1496]: As long as we're fixing typos ("checked" -> "check") I'd like to see "can not" corrected to "cannot".  ("Can not" is properly used when the "not" is part of a separate phrase -- for example, "I can check out the run, or I can not check out the run; you tell me which you prefer I should do".  There's no separate phrase here to which "not" applies.)
But this is obviously totally trivial in terms of the real issue -- the wrong variable being printed.

File: src/edu/csus/ecs/pc2/core/model/InternalContest.java [hunk 2]
```diff
@@ -2258,7 +2258,7 @@ public Clarification checkoutClarification(Clarification clar, ClientId whoChang
                 clarificationList.updateClarification(newClar);
                 return clarificationList.get(clar.getElementId());
             } else {
-                throw new ClarificationUnavailableException("Client " + clientId + " can not checked out clar " + clar.getNumber() + " (site " + clar.getSiteNumber() + ")");
+                throw new ClarificationUnavailableException("Client " + whoChangedClar + " can not check out clar " + clar.getNumber() + " (site " + clar.getSiteNumber() + ")");
             }
```
Comment (clevengr) [inline src/edu/csus/ecs/pc2/core/model/InternalContest.java:2262]: Ditto on "can not".

**Review by clevengr — APPROVED** [review]
My trivial-issue comments not withstanding, I "cannot" find a reason not to approve this PR.

**Review by lane55 — APPROVED** [review]
I thought I can not approve - but I do.

**Pull request merged**